auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/internal/AdminUserCreationController.kt

```diff
@@ -66,6 +67,25 @@ class AdminUserCreationController(
                 "This email is invalid. Please check it was entered correctly"
             )
         }
+        val userPermissions = try {
+            deltaUserPermissionsRequestMapper.deltaRequestToUserPermissions(deltaUserDetailsRequest)
+        } catch (e: DeltaUserPermissions.Companion.BadInput) {
+            throw ApiError(
+                HttpStatusCode.BadRequest,
+                "bad_request",
+                e.message!!,
+                "Requested permissions are invalid: ${e.message}"
+            )
+        }
+        if (userPermissions.roles.contains(DeltaSystemRole.ADMIN)) {
+            throw ApiError(
+                HttpStatusCode.UnprocessableEntity,
+                "admin_role_forbidden",
+                "Cannot assign admin role",
+                "Cannot assign admin role",
+            )
+        }
```

Comment:
`AdminEditUserController` handles it separately. It also needs to not

Reply:
> And is there any reason for it to be here rather than just being another cause of `deltaRequestToUserPermissions` throwing DeltaUserPermissions.Companion.BadInput?

I think the only downside would be it would then reject requests to edit a user that sends the admin role for a user which is already an admin, currently it would just be ignored. I don't think we do that at the moment anyway though. Do you think it's worth changing?

Reply:
Nah, that's fine. I didn't notice the admin role check in the edit controller.